mypy.ini

```diff
@@ -39,10 +39,25 @@ ignore_missing_imports = True
 [mypy-httpx.*]
 ignore_missing_imports = True
 
+# Don't check LSST packages generally or even try to import them, since most
+# don't have type annotations.
+
 [mypy-lsst.*]
 ignore_missing_imports = True
 ignore_errors = True
 
+# Do expect to import type annotations from LSST middleware dependencies that
+# should have them, but don't re-check them as they have their own mypy.ini
+# files with their own exceptions.
+
+[mypy-lsst.utils.*]
+ignore_missing_imports = False
+ignore_errors = True
```

Comment:
Why didn't we need this before? Did the previous block mean that we weren't testing utils/resources at all before? That can't be right because when I factored out the utility code I did get mypy errors.

Reply:
It means that when `daf_butler` imported types from `utils` or `resources` before, they were all effectively seen as `Any` by MyPy, so how much it could tell us about our usage of them was pretty limited.